include/mls/messages.h

```diff
@@ -527,6 +594,83 @@ struct MLSPlaintext
   bool decrypted;
 };
 
+struct MLSPlaintextCommitContent
+{
+  const bytes& group_id;
+  const epoch_t& epoch;
+  const Sender& sender;
+  const bytes& authenticated_data;
+
+  const ContentType content_type{ ContentType::commit };
+  const Commit& commit;
+
+  const bytes& signature;
+
+  TLS_SERIALIZABLE(group_id,
+                   epoch,
+                   sender,
+                   authenticated_data,
+                   content_type,
+                   commit,
+                   signature)
+  TLS_TRAITS(tls::vector<1>,
+             tls::pass,
+             tls::pass,
+             tls::vector<4>,
+             tls::pass,
+             tls::pass,
+             tls::vector<2>)
+};
+
+struct MLSPlaintextCommitAuthData
+{
+  const std::optional<MAC>& confirmation_tag;
+
+  TLS_SERIALIZABLE(confirmation_tag)
+  TLS_TRAITS(tls::pass)
+};
+
+// struct {
+//     select (MLSCiphertext.content_type) {
+//         case application:
+//           opaque application_data<0..2^32-1>;
+//         case proposal:
+//           Proposal proposal;
+//         case commit:
+//           Commit commit;
+//     }
+//     opaque signature<0..2^16-1>;
+//     optional<MAC> confirmation_tag;
+//     opaque padding<0..2^16-1>;
+// } MLSCiphertextContent;
+struct MLSCiphertextContent
+{
+  const var::variant<ApplicationData, Proposal, Commit>& content;
+  const bytes& signature;
+  const std::optional<MAC>& confirmation_tag;
+  const bytes& padding;
```

Comment:
Since the padding is always zero and (b) we have a custom `operator<<` anyway, we could just have `const size_t padding_size` and call `encoded.resize(encoded.size() + padding_size, 0)` to add the pading.